desugarer.go

```diff
@@ -252,9 +252,31 @@ func desugarArrayComp(comp *ast.ArrayComp, objLevel int) (ast.Node, error) {
 	return desugarForSpec(wrapInArray(comp.Body), &comp.Spec)
 }
 
-func desugarObjectComp(astComp *ast.ObjectComp, objLevel int) (ast.Node, error) {
-	return &ast.LiteralNull{}, nil
-	// TODO(sbarzowski) this
+func desugarObjectComp(comp *ast.ObjectComp, objLevel int) (ast.Node, error) {
+
+	// TODO(sbarzowski) find a consistent convention to prevent desugaring the same thing twice
+	// here we deeply desugar fields and it will happen again
+	err := desugarFields(*comp.Loc(), &comp.Fields, objLevel+1)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(comp.Fields) != 1 {
+		return nil, fmt.Errorf("Too many fields in object comprehension, it should have been caught during parsing")
```

Comment:
If you're breaking an internal contract it might be best to `panic`

Reply:
right, done